packages/website/ts/components/inputs/allowance_toggle.tsx

```diff
@@ -74,7 +75,24 @@ export class AllowanceToggle extends React.Component<AllowanceToggleProps, Allow
             newAllowanceAmountInBaseUnits = DEFAULT_ALLOWANCE_AMOUNT_IN_BASE_UNITS;
         }
         try {
-            await this.props.blockchain.setProxyAllowanceAsync(this.props.token, newAllowanceAmountInBaseUnits);
+            await this.props.blockchain.setProxyAllowanceAsync(this.props.token, newAllowanceAmountInBaseUnits).then(
+                () => {
+                    ReactGA.event({
+                        category: 'Portal',
+                        action: 'Set Allowance Success',
+                        label: this.props.token.symbol,
+                        value: newAllowanceAmountInBaseUnits.toNumber(),
+                    });
+                },
+                () => {
+                    ReactGA.event({
+                        category: 'Portal',
+                        action: 'Set Allowance Failure',
+                        label: this.props.token.symbol,
+                        value: newAllowanceAmountInBaseUnits.toNumber(),
+                    });
+                },
+            );
```

Comment:
No need to use `.then` here, we can just call these two calls after we've finished awaiting the `setProxyAllowanceAsync` call. Any specific reason you used promises here?